ql/experimental/credit/riskybond.hpp

```diff
@@ -71,6 +83,11 @@ namespace QuantLib {
         //@{
         bool isExpired() const;
         //@}
+        //! resets engine's npv date.
+        void setNpvDate(const Date d) { 
```

Comment:
This is required because `npvDate_` is set to `Settings::instance().evaluationDate()` at construction time.  It could be left equal to `Date()` and checked at the beginning of the calculations instead.

Reply:
I'm  not sure I understand this suggestion. If we get rid of this and adjust npvDate_ to evaluationDate inside performCalculations, we still can't change the npvDate_ to something > than evaluationDate.

Reply:
My idea was that instead of having
```
npvDate_(npvDate == Date() ? Settings::instance().evaluationDate() : npvDate)
```
in the constructor (which fixes the NPV date, so that we're then forced to change it manually), we can save `npvDate_` as it is, even if null.  If it's a valid date, we use it in the calculations.  If it's null, we use the evaluation date in the calculations (but we leave the data member null, so that if we change the evaluation date and recalculate the NPV date changes as well).  This would work even if we wanted the keep the NPV date, say, two days after the evaluation date; we can store a number of settlement days and calculate the NPV date on the fly before the calculation.  Did you have other use cases in mind?

Reply:
No, the evaluation date stays whatever it is, because it affects everything as you say. A local `npvDate` variable gets set to the evaluation date plus settlement and used in the calculation.

Reply:
I don't see where you would get the settlement days though, from rateTs? Sorry for being slow... Please make the change in the code, maybe I'll get it then:)

Reply:
They would be passed instead of the NPV date.  I'll make the change.

Reply:
Hi, sorry for the late reply.
As for my intention, I guess it was to be able to change the pricing date -think VaR etc- (sorry for stating what you all thought) in a,yes, possibly untidy way. 
If I understood the problem, why not writing a separate engine? and register with the eval date.